benches/throughput.rs

```diff
@@ -374,12 +380,139 @@ pub fn metrics(c: &mut Criterion) {
     });
 }
 
+/// Iterate through possible IP pairs restricted to 0 < ip_pair.0 < ip_pair.1 <= MAX_HBONE_WORKLOADS.
+fn next_ip_pair(ip_pair: (u8, u8)) -> (u8, u8) {
+    if ip_pair.0 == 0 || ip_pair.1 == 0 {
+        panic!("Invalid host");
+    }
+    match (
+        Ord::cmp(&ip_pair.0, &(MAX_HBONE_WORKLOADS - 1)),
+        Ord::cmp(&ip_pair.1, &MAX_HBONE_WORKLOADS),
+    ) {
+        (Greater, _) | (_, Greater) | (Equal, Equal) => panic!("Invalid host"),
+        (_, Less) => (ip_pair.0, ip_pair.1 + 1),
+        (Less, Equal) => (ip_pair.0 + 1, ip_pair.0 + 2),
+    }
+}
+
+/// Reserve IPs in 127.0.1.0/24 for these HBONE connection tests.
+/// Thus, we identify hosts by a u8 in the form 127.0.1.x
+fn hbone_connection_ip(x: u8) -> IpAddr {
+    IpAddr::V4(Ipv4Addr::new(127, 0, 1, x))
+}
+
+fn hbone_connection_config() -> ztunnel::config::ConfigSource {
+    let mut workloads: Vec<LocalWorkload> = Vec::with_capacity(MAX_HBONE_WORKLOADS as usize);
+    // We can't create one work load with many IPs because ztunnel could connect to any one causing
+    // inconsistent behavior.
```

Comment:
You could also clarify in the comment what approach your *are* taking. 

`Instead, we create many workloads with unique IPs to ensure a new HBONE connection..."